Day 4 게시글 생성시간 저장 및 게시글 검색 기능 구현

leerobing opened the pull request:
예외 발생할 때 응답으로 예외 메세지만 내려보내고 있는데 이렇게 되면 응답을 받는 쪽에서 예외에 대한 정보를 알수가 없어 좋지 않을거 같습니다. 예외가 발생했음을 응답으로 내려보낼때 좋은 규격? 좋은 형식이 있을까요?

leerobing committed:
feat: 게시글 생성시간 저장 및 게시글 검색 기능 구현

leerobing committed:
feat: 유효성 검사 및 예외 처리 구현

leerobing committed:
refactor: 예외 처리 및 응답 기능 수정

예외 발생 시 응답으로 내려 보내는 객체 안에 성공 여부와 해당 예외에 대한 코드, 메세지를 담았다.

leerobing committed:
refactor: 응답 dto 중복 제거

Response 객체를 따로 만들었기 때문에 Board 타입의 객체를 ResponseDto에 또 넣을 필요가 없다고 판단하여 바로 Board 타입 객체를 Response 객체에 넣어줬다.

kkangh00n commented:
감사합니다 ㅎㅎ

jaeseongDev commented:
질문 잘 해주셨습니다~!! 응답을 할 때에 일반적으로 많이 포함하는 정보는 '상태코드, 에러코드, 에러 메시지 내용' 이렇게 3가지 입니다. 여기서 이 3가지를 왜 다 포함하는 게 좋은 지를 공부해보시면 좋을 것 같아요~!! '굳이 이건 포함 안 해도 되지않나?', '이건 왜 포함하는 거지?'라는 질문을 스스로 해보시기 바랍니다:)

leerobing committed:
refactor: 예외처리 응답 객체 팩토리 메서드 패턴 적용

기존에는 새로운 객체를 만들시 new 키워드를 사용하여 새로운 객체를 만들고 속성에 하나씩 값을 대입한 후 응답 객체를 반환했는데 팩토리 메서드를 사용하여 객체를 생성한 후 반환하게 수정하였다.
